### PR TITLE
Feature/clean unused state props 2009

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/api/repositories/agoraRepository/agoraRepository.api.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {generatePath} from 'react-router-dom';
+
+import {RequestInterface} from 'api/interfaces';
+import {request} from 'api/request';
+
+import {agoraRepositoryApiEndpoints} from './agoraRepository.api.endpoints';
+import {AgoraTokenRequest, AgoraTokenResponse} from './agoraRepository.api.types';
+
+export const getAgoraToken: RequestInterface<AgoraTokenRequest, AgoraTokenResponse> = (options) => {
```
Comment (PAKovalev) [inline src/api/repositories/agoraRepository/agoraRepository.api.ts:9]: Can we create two functions? getAgoraToken & getAgoraScreenShareToken

File: src/api/repositories/spaceIntegrationsRepository/spaceIntegrations.api.types.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  };
+  integrationTypeId: {
+    type: string;
+    data: Buffer;
+  };
+  data?: IntegrationDataInterface;
+}
+
+export interface SpaceIntegrationsStageModeResponse
+  extends Omit<SpaceIntegrationInterface, 'spaceId'> {
```
Comment (PAKovalev) [inline src/api/repositories/spaceIntegrationsRepository/spaceIntegrations.api.types.ts:57]: Why?

File: src/core/models/Space/Space.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import {UUIDModel} from 'core/models/UUID';
+import {SpaceAncestorModel} from 'core/models/SpaceAncestor';
+import {bytesToUuid} from 'core/utils';
+import {SpaceUserModel} from 'core/models/SpaceUser';
+import {SubSpaceModel} from 'core/models/SubSpace';
+import {RequestModel} from 'core/models/Request';
+import {ResetModel} from 'core/models/ResetModel';
+import {api, SpaceResponse} from 'api';
+import {SpaceType} from 'core/enums';
+import {GetAllowedSpaceTypesResponse} from 'api/repositories/spaceTypeRepository/spaceTypeRepository.api.types';
```
Comment (PAKovalev) [inline src/core/models/Space/Space.ts:12]: `import {GetAllowedSpaceTypesResponse} from 'api';`

File: src/core/models/Space/Space.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+      subSpaces: types.optional(types.array(SubSpaceModel), []),
+      isStakeShown: false,
+      allowedSpaceTypes: types.optional(types.array(types.string), []),
+
+      // Other
+      kusamaOperatorId: types.maybe(types.string)
+    })
+  )
+  .actions((self) => ({
+    setup(spaceId: string, isTable = false) {
```
Comment (PAKovalev) [inline src/core/models/Space/Space.ts:54]: unused function

File: src/scenes/collaboration/Collaboration.tsx [hunk 5]
```diff
@@ -1,156 +1,149 @@
...
-  const {unityStore} = mainStore;
+  const {collaborationStore, mainStore, sessionStore} = useStore();
+  const {unityStore, agoraStore} = mainStore;
+
+  const {addRequestPopup} = useStageModePopupQueueContext();
 
   const {spaceId} = useParams<{spaceId: string}>();
   const {t} = useTranslation();
   const history = useHistory();
+  const [isModerator, , ,] = useModerator(spaceId);
```
Comment (PAKovalev) [inline src/scenes/collaboration/Collaboration.tsx:32]: Can we refactor?

File: src/scenes/collaboration/Collaboration.tsx [hunk 6]
```diff
@@ -1,156 +1,149 @@
...
-  const {collaborationState, collaborationDispatch} = useCollaboration();
-  const stageModeState = useStageModeStatusInfo(collaborationState.collaborationSpace?.id);
   const switchDeviceModal = useRef<ModalRef>(null);
   const [newDevice, setNewDevice] = useState<MediaDeviceInfo>();
-  const agoraStageMode = useAgoraStageMode();
-  const joinMeetingSpace = useJoinCollaborationSpaceByAssign();
-  const stageModeJoin = useStageModeJoin(collaborationState.collaborationSpace?.id);
-  const stageModeLeave = useStageModeLeave(collaborationState.collaborationSpace?.id);
+
+  useEffect(() => {
```
Comment (PAKovalev) [inline src/scenes/collaboration/Collaboration.tsx:37]: Remove. It is in UnityPage.

File: src/scenes/collaboration/Collaboration.tsx [hunk 7]
```diff
@@ -1,156 +1,149 @@
...
+    unityStore.pause();
+
+    return unityStore.resume;
+  }, [unityStore]);
+
+  useEffect(() => {
```
Comment (PAKovalev) [inline src/scenes/collaboration/Collaboration.tsx:43]: Can we move to RootStore? + leaveMeeting?

File: src/scenes/collaboration/pages/CalendarPage/CalendarPage.tsx [hunk 8]
```diff
@@ -11,53 +11,37 @@ import {absoluteLink} from 'core/utils';
...
 
-  // TODO: Refactor legacy hooks to mobx
-  const {collaborationState, collaborationDispatch} = useCollaboration();
-  const leaveCollaborationSpaceCall = useLeaveCollaborationSpace();
-  const stageModeLeave = useStageModeLeave(collaborationState.collaborationSpace?.id);
-
-  // TODO: make as reusable action in store
-  const leaveCollaborationSpace = () => {
-    if (collaborationState.collaborationSpace) {
+  const handleClose = async () => {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/CalendarPage/CalendarPage.tsx:29]: Move to RootStore

File: src/scenes/collaboration/stores/RootCollaborationStore.ts [hunk 9]
```diff
@@ -1,25 +1,87 @@
...
+  )
+  .volatile(() => ({
+    leftMeetingTimer: setTimeout(() => {}, 0)
+  }))
   .actions((self) => ({
-    init(spaceId: string) {
-      self.spaceStore.setSpace(spaceId);
-      self.spaceStore.fetchSpaceInformation();
-      self.dashboardStore.dashboard.fetchDashboard(spaceId);
+    joinMeetingSpace: flow(function* (spaceId: string, isTable = false) {
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/RootCollaborationStore.ts:35]: Move to RootStore

File: src/scenes/collaboration/stores/RootCollaborationStore.ts [hunk 10]
```diff
@@ -1,25 +1,87 @@
...
+        throw new PrivateSpaceError();
+      }
+
+      yield self.space.fetchSpaceInformation();
+
+      if (!isTable) {
+        yield self.dashboard.fetchDashboard(spaceId);
+      }
+
+      UnityService.triggerInteractionMsg?.(PosBusEventEnum.EnteredSpace, spaceId, 0, '');
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/RootCollaborationStore.ts:50]: Use store after moving

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/api/repositories/spaceIntegrationsRepository/spaceIntegrations.api.types.ts (hunk 2 above)
Comment (Daboo1998) [reply src/api/repositories/spaceIntegrationsRepository/spaceIntegrations.api.types.ts:57]: Cause the extended interface has spaceId of different type, and didn't want to redeclare other fields again. Its bacause of BE, we need to also refactor BE EP's :)

File: src/scenes/collaboration/Collaboration.tsx (hunk 5 above)
Comment (Daboo1998) [reply src/scenes/collaboration/Collaboration.tsx:32]: Yes, we can! :) 

unless you want it to be part of a different ticket

File: src/api/repositories/agoraRepository/agoraRepository.api.ts (hunk 1 above)
Comment (PAKovalev) [reply src/api/repositories/agoraRepository/agoraRepository.api.ts:9]: 2 will better.

File: src/scenes/collaboration/Collaboration.tsx (hunk 5 above)
Comment (PAKovalev) [reply src/scenes/collaboration/Collaboration.tsx:32]: It is time to refactor )

Comment (Daboo1998) [reply src/scenes/collaboration/Collaboration.tsx:32]: Ok 👍

File: src/component/molucules/TopBar.tsx [hunk 11]
```diff
@@ -86,38 +73,32 @@ const TopBar = ({
...
     }
   };
 
   const isAdminShown = () => {
-    return currentLocation.pathname.includes('/space/' + spaceId + '/admin');
+    if (!space) {
+      return false;
+    }
+
+    return currentLocation.pathname.includes('/space/' + space.id + '/admin');
```
Comment (aminbksi) [inline src/component/molucules/TopBar.tsx:101]: use Routes maybe?

Comment (PAKovalev) [reply src/component/molucules/TopBar.tsx:101]: We have to remove this component.

File: src/component/overlays/VideoLayer.tsx [hunk 12]
```diff
@@ -1,10 +1,9 @@
-import React, {useState} from 'react';
-import {useHistory} from 'react-router-dom';
+import React, {useCallback, useState} from 'react';
+import {generatePath, useHistory} from 'react-router-dom';
 
 import {useUnityEvent} from 'shared/hooks';
 
 import DashboardVideo from '../atoms/video/DashboardVideo';
-import {useJoinCollaborationSpaceByAssign} from '../../context/Collaboration/hooks/useCollaboration';
 import {ROUTES} from '../../core/constants';
```
Comment (aminbksi) [inline src/component/overlays/VideoLayer.tsx:7]: path

**Review by aminbksi — CHANGES_REQUESTED** [review]

File: src/core/models/Space/Space.ts (hunk 3 above)
Comment (Daboo1998) [reply src/core/models/Space/Space.ts:12]: For that I would have to add `export * from './spaceTypeRepository.api.types';` to `api/repositories/spaceTypeRepository/index.ts` and `import * from './repositories';` to `api/index.ts`. We never did that, do you want to do that?

File: src/scenes/collaboration/Collaboration.tsx (hunk 6 above)
Comment (Daboo1998) [reply src/scenes/collaboration/Collaboration.tsx:37]: I added it cause it didn't pause unity when loading from url. Are you sure its done correctly in UnityPage? Or is it a recent change?

File: src/scenes/collaboration/pages/CalendarPage/CalendarPage.tsx (hunk 8 above)
Comment (Daboo1998) [reply src/scenes/collaboration/pages/CalendarPage/CalendarPage.tsx:29]: I will even remove the calls, going to ROUTES.base should trigger the useEffect return from Collaboration.tsx

File: src/scenes/collaboration/stores/RootCollaborationStore.ts (hunk 9 above)
Comment (Daboo1998) [reply src/scenes/collaboration/stores/RootCollaborationStore.ts:35]: No I will just call it from RootStore and call the unity stuff from there

File: src/component/molucules/TopBar.tsx (hunk 11 above)
Comment (Daboo1998) [reply src/component/molucules/TopBar.tsx:101]: So I will leave it for now

File: src/component/overlays/VideoLayer.tsx (hunk 12 above)
Comment (Daboo1998) [reply src/component/overlays/VideoLayer.tsx:7]: It wasn't't my change but I can change that

**Review by PAKovalev — APPROVED** [review]

**Review by aminbksi — APPROVED** [review]

**Pull request merged**